src/plugins/data_source/server/saved_objects/credential_saved_objects_client_wrapper.ts

```diff
@@ -138,15 +139,14 @@ export class CredentialSavedObjectsClientWrapper {
   }
 
   private async validateAndEncryptPartialAttributes<T = unknown>(attributes: T) {
-    this.validateCredentialMaterials(attributes.credentialMaterials);
+    this.validateAttributes(attributes);
```

Comment:
can we check password attributes here instead of down streams?

For example, use `in` type guard

```
  if ("password" in credentialMaterialsContent) 
```